style/react/README.md

```diff
@@ -0,0 +1,50 @@
+# React
+
+ * Use React in [Strict Mode]
+ * Use React with [TypeScript](../typescript/README.md)
+ * Use [Jest](../testing-jest/README.md) for unit testing
+ * Use [styled-components] for CSS-in-JS
```

Comment:
I'm pretty ambivalent about styled components vs scss modules. The `Use` here would be _use this if you need css-in-js_.

It might be the case that we never want to encourage css-in-js, in which case I would be in favor of removing this 🙂 

Some of these specific technology recommendations are also moving to templates in https://github.com/thoughtbot/react-suspenders/pull/1